docs/instructor-training-workflow.txt

```diff
@@ -0,0 +1,211 @@
+## Notation
+
+- [use cases] are in square brackets
+- {web pages} are in curly braces
+- <database objects> are in angle brackets
+
+## Individual application for training.
+
+* [user registers self][a]: user match by email address or GitHub ID.
+  - User fills in {single user application form} on AMY with name, email address, GitHub ID, etc.
+  - AMY matches to pre-existing <Person> using email or GitHub ID.
+  - <Training Application> created, connected to pre-existing <Person>, marked "Pending", and added to <Training Application List>.
+
+* [user registers self][b]: no user match.
+  - User fills in form on AMY with name, email address, GitHub ID, etc.
+  - AMY does *not* match email or GitHub ID.
+  - <Training Application> created, marked "Pending", and added to <Training Application List> *without* <Person> object.
+
+## Administrator approves individual for training.
+
+* [admin approves training][a]: user record already in database.
+  - Admin views <Training Application List>.
+  - Admin selects <Training Application> marked "Pending" that has been matched with <Person>.
+  - Admin approves <Training Application>.
+  - AMY clears <Training Application>'s "Pending" flag.
+
+* [admin approves training][b]: no user record in database.
+  - Admin views <Training Application List>.
+  - Admin selects <Training Application>.
```

Comment:
I think we need to approve applications one by one, because some will have Person objects and some won't.  If this works, we can go back later and add something to the application list display to allow bulk approval.